core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushAggregationIntoTableScan.java

```diff
@@ -225,7 +225,8 @@ private static AggregateFunction toAggregateFunction(Context context, Aggregatio
                 .map(symbol -> new Variable(symbol.getName(), context.getSymbolAllocator().getTypes().get(symbol)));
 
         return new AggregateFunction(
-                signature.getName(),
+                metadata.getFunctionMetadata(aggregation.getResolvedFunction())
+                        .getResolvedName(),
```

Comment:
Instead of carrying this name throughout the planning process, the function should be resolved to it's canonical signature during initial planning.

Reply:
Thanks for your comments ! 😃 

I figure out a way, but not sure if this is appropriate. The way is to compare the origin name and the canonical name in Visitor.visitFunctionCall in ExpressionAnalyzer.java, if not equals, resolve the function to which has canonical name.

before
https://github.com/trinodb/trino/blob/cf68459a0d5d114d75bbb0897bca54654ae21a14/core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java#L1012
after
```java
function = metadata.resolveFunction(node.getName(), argumentTypes);

FunctionMetadata functionMetadata = metadata.getFunctionMetadata(function);
String originName = functionMetadata.getSignature().getName();
String canonicalName = metadata.getFunctionMetadata(function).getResolvedName();
if (!originName.equals(canonicalName)) {
    function = metadata.resolveFunction(QualifiedName.of(canonicalName), argumentTypes);
}
```